ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala

```diff
@@ -257,8 +257,8 @@ private[state] object Conversions {
       }.toMap
     )
 
-  private def encodeParties(parties: Iterable[Party]): List[String] =
-    parties.asInstanceOf[Set[String]].toList.sorted
+  private def encodeParties(parties: Set[Party]): List[String] =
+    (parties.toList: List[String]).sorted
```

Comment:
`Set` is invariant, so we cannot use the type ascription on it directly. But since we anyway convert to a `List`, we can do it there.